lib/engine/game/g_18_mag/game.rb

```diff
@@ -616,31 +619,47 @@ def check_distance(route, visits)
           raise GameError, 'Must visit minimum of two non-mine stops' if visits.sum(&:visit_cost) < 2
         end
 
+        def compute_stops(route, train = nil)
+          train ||= route.train
+          distance = if gc_train?(route) && !other_gc_train?(route)
+                       gc_train_distance(route.train.distance)
+                     else
+                       route.train.distance
+                     end
+          train_dup = train.dup
+          train_dup.distance = distance
+
+          super(route, train_dup)
+        end
+
         def gc_train_distance(route_distance)
           if route_distance.is_a?(Numeric)
             town_distance_value = route_distance
+            city_distance_value = route_distance
           else
             # route has a 1 town value from the supporter
             town_distance = route_distance.find { |n| n['nodes'] == ['town'] }
-            town_distance_value = town_distance['pay'] + route_distance
+            city_distance_value = route_distance.find { |n| n['nodes'].include?('city') }['pay']
+            town_distance_value = town_distance['pay'] + city_distance_value
           end
           [
-              {
-                nodes: %w[city offboard town],
-                pay: route_distance,
-                visit: route_distance,
-              },
-              {
-                nodes: %w[town],
-                pay: town_distance_value,
-                visit: town_distance_value,
-              },
+            {
+              nodes: %w[town],
+              pay: town_distance_value,
+              visit: town_distance_value,
+            },
+            {
+              nodes: %w[city offboard town],
+              pay: city_distance_value,
+              visit: city_distance_value,
```

Comment:
I found the source of the validation errors, the classic server/client string/symbol issue. The base method expects the keys in these `distance` hashes to be strings: https://github.com/tobymao/18xx/blob/6310b539085311620e4382d7ec38958def286e7e/lib/engine/game/base.rb#L1373

In the compiled JS (client side / hotseat mode), symbols and strings are equivalent, but in the server side Ruby (and in `validate.rb`) they are not, so you need to use strings instead of symbols:

```suggestion
              'nodes' => %w[town],
              'pay' => town_distance_value,
              'visit' => town_distance_value,
            },
            {
              'nodes' => %w[city offboard town],
              'pay' => city_distance_value,
              'visit' => city_distance_value,
```